lib/eslint/eslint.js

```diff
@@ -271,19 +281,47 @@ function findFlatConfigFile(cwd) {
     );
 }
 
+/**
+ * Check if the file is a TypeScript file.
+ * @param {string} filePath The file path to check.
+ * @returns {boolean} `true` if the file is a TypeScript file, `false` if it's not.
+ */
+function isFileTS(filePath) {
+    const fileExtension = path.extname(filePath);
+
+    return /^\.[mc]?ts$/u.test(fileExtension);
+}
+
+/**
+ * Check if ESLint is running in Bun.
+ * @returns {boolean} `true` if the ESLint is running Bun, `false` if it's not.
+ */
+function isRunningInBun() {
+    return !!globalThis.Bun || !!globalThis.process?.versions?.bun;
```

Comment:
I'm not sure tbh but [here](https://bun.sh/guides/util/detect-bun) is where I got that from.

Reply:
From that doc:

> In TypeScript environments, the previous approach will result in a type error unless `bun-types` is globally installed. To avoid this, you can check `process.versions` instead.

ESLint isn't developed in TypeScript, so the `process.versions` check -which is only mentioned to help appease the type checker- is unnecessary.

```suggestion
    return !!globalThis.Bun;
```

Reply:
Wow, I honestly don't know how I missed that. Thank you so much for letting me know.